scheduler/pkg/coordinator/hub.go

```diff
@@ -68,6 +68,12 @@ func NewEventHub(l log.FieldLogger) (*EventHub, error) {
 	return &hub, nil
 }
 
+func (h *EventHub) IsClosed() bool {
+	h.lock.RLock()
+	defer h.lock.RUnlock()
+	return h.closed
```

Comment:
Right now, the hub only becomes closed from a `defer` statement in `main.go`. So the check is not strictly necessary, but we can keep it for completeness.